hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/BasicClusterStateTest.java

```diff
@@ -424,8 +425,7 @@ private void pendingInvocations_shouldBeNotified_whenMemberLeft_whenClusterState
         changeClusterStateEventually(hz2, state);
         hz1.shutdown();
 
-        expectedException.expect(MemberLeftException.class);
-        future.get();
+        assertTrueEventually(() -> Assert.assertThrows(MemberLeftException.class, future::get));
```

Comment:
Why this fixes the issue? If the first `get()` call completes with `IllegalStateException`, all subsequent ones will return the same exception, no? WDYT about making `SilentOperation` `AllowedDuringPassiveState` or asserting eventually that cluster isn't in passive state before calling `future.get()`?

Reply:
That exception returns from the `hz1`, and it means that it is still available, but currently in a Passive state before the ShuttingDown. So, I assume we should wait for `hz1` to be completely shut down.

Reply:
We can implement the `AllowedDuringPassiveState` but in that case, we also need some eventual waiting when the `hz1` will be shut down to get `MemberLeftException`

Reply:
> That exception returns from the `hz1`

Ok now I understand.